services/client/src/withPagination.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import WayPoint from "react-waypoint";
+import Spinner from "components/spinner/loading";
+
+const withPagination = url => Component => {
+  class Pagination extends React.Component {
+    state = {
+      page: 1,
+      pageSize: 50
+    };
```

Comment:
Why aren't we getting these values from props? Shouldn't this state be in the store?

Reply:
This is intentionally here to avoid writing this piece of state for each resource in the store. If you think to paginate speakers, videos or another page you don't have to write it again. What do you think of this approach?

Reply:
I think it could be worth getting these values from props; `events` are pretty large objects that we don't make full use of yet however other objects such as speakers or videos contain much fewer data. We will likely want the ability to play with the pagination for each.

Reply:
`pageSize` probably isn't something that'll change, so it may just be a hard coded value in the container (and as Kyle says, vary between what's being requested), vs the dynamic `page` value which will be updated on each successful request.

Keeping values in local state has bitten me in the past and I think in this case we probably want the page requested in the store. One example I can think of is using client-side routing to switch between the events page to video page and back agin - we'd want to continue with the list of events that were originally displayed, but because this component was unmounted we'd lose the state of which page we last requested.

Reply:
I do appreaciate the review guys! I'll set this data into `store.events` and the `pageSize` value as a constant into the container. @alistairjcbrown thanks for sharing the experience and the use case.